simulation.py

```diff
@@ -298,22 +327,34 @@ def mono_bi_edge_phase(self, phase):
         edge_matrix = self.edge_matrix_phase[phase]
         return edge_matrix[0][0] + edge_matrix[1][1], edge_matrix[0][1]
 
-    def gini_coeff(self, grp=None, idx_list=None):
+    def gini_coeff(self, grp=None, idx_list=None, ab_naive=True):
         degree_list = list(self.G.degree())
-        if grp:
+        if grp == 'treatment':
+            degree_list = [(n,d) for n, d in degree_list if n in self.get_treatment_nodes()]
+        elif grp == 'control':
+            if ab_naive:
+                degree_list = [(n,d) for n, d in degree_list if n in self.get_control_nodes()]
+            else:
+                degree_list = [(n,d-self.G.nodes[n]['num_rec']) for n, d in degree_list if n in self.get_control_nodes()]
+        elif grp:
             color = grp.color
             degree_list = [(n,d) for n, d in degree_list if self.G.nodes[n]['color'] == color]
-        if idx_list:
+        elif idx_list:
             degree_list = [(n,d) for n, d in degree_list if n in idx_list]
         degree_list = [d for _, d in degree_list]
         degree_list = np.sort(degree_list)
         n = len(degree_list)
+        if n == 0:
+            return 0
         gini = 2 * np.sum( np.arange(n) * degree_list) / (n * np.sum(degree_list)) - (n + 1) / n
         return gini
         
-    def homophily(self, grp: Group):
+    def homophily(self, grp, ab_naive=True):
         color = grp.color
-        return self.edge_matrix[color][color] / np.sum(self.edge_matrix, axis=0)[color] - grp.size / self.nodes
+        if ab_naive:
+            return self.edge_matrix[color][color] / np.sum(self.edge_matrix, axis=0)[color] - grp.size / self.nodes
+        else:
+            return (self.edge_matrix[color][color] - self.edge_matrix_3[color][color]) / (np.sum(self.edge_matrix, axis=0)[color] - np.sum(self.edge_matrix_3, axis=0)[color]) - grp.size / self.nodes
```

Comment:
here lets not substract ` self.edge_matrix_3[color][color]` (it should be 0) for the control group either way!

Reply:
done!